add a mission feasibility check section (#2226)
PX4/PX4-user_guide: en/flying/missions.md

![planning-mission](../../assets/flying/planning_mission.jpg)

### Mission Feasibility Checks

PX4 runs some basic sanity checks to determine if a mission is feasible.
For example, whether the mission is close enough to the vehicle, if the mission will conflict with a geofence, or if a mission landing pattern is required but is not present.

> The checks are run when it is uploaded and when it is first armed.
> If any of the checks fail, the user is notified and it is not possible to start the mission (on upload the mission may not even be saved).

Reviewer: ```suggestion
The checks are run when the mission is uploaded and immediately before it is run.
If any of the checks fail, the user is notified and it is not possible to start the mission.

```

![planning-mission](../../assets/flying/planning_mission.jpg)

> ### Mission Feasibility Checks

Reviewer: I'm not sure if this section is better here or in http://docs.px4.io/main/en/flight_modes/mission.html. 

Generally we use this mission planning page for planning - it has always been a bit "thin" because we assume that QGC tells you most of what you need to know - i.e. what commands you can use. It is only recently I have expanded it a bit. We use the flight mode page (http://docs.px4.io/main/en/flight_modes/mission.html) for more detailed information that you might use both for manually planning missions and to understand the mechanics / debugging / companion missions etc.

I think that where you have put it is the right place since you probably should know this if you're planning a mission. But then probably you also need it cross linked from the mode page?

Note, I need to put some hard thought into this whole page. Never been a priority because most of the time you can pick up what you need to know in QGC quite easily.

Author: >My leaning is to put this in http://docs.px4.io/main/en/flight_modes/mission.html and have a small section in this page.

Ah yes, that makes sense. The checks are clearly a "PX4 thing" and thus should live in the flight mode section. And here only reference that one, and mention that there are checks being run at mission upload/vehicle boot that need to pass to fly the mission. But really keep the flying/mission page to the bare minimum. 

The link from the flying/mission page to flight_modes/mission is missing atm.

>what are we hoping a user to know that they have to do when mission planning with this information?

I think it is important for users to know that there is some mission checking going on, and give them roughly an idea what the checks involve. Plus what to change if they have failures.

Reviewer: @sfuhrer I reduced the section to be "high level" and linked.

> Upon receiving an updated mission file from the ground station, the flight controller runs some feasibility checks that are meant to ensure the safety of the vehicle and catch unintended mission settings. If one of the checks fails, the user is notified and it is not possible to start the mission.

Reviewer: My understanding is that checks are run on more than just upload. I.e. I could upload a mission and the vehicle fails the checks, I could then drive to the place where the mission is supposed to be used and it passes the checks.

Point being that the checks must be run again at some point - I am assuming they are run every time you switch to mission mode? We should make it clear.

Author: True, the checks are also run once after boot as soon as the home position is valid. Switching to Mission doesn't seem to be necessary.

I'm though not even sure if a rejected mission is stored on the vehicle, so if your example of the checks failing at location A, then passing automatically at location B would work. 

@Igor-Misic  weren't you involved in the building the logic around there? Would you know if this example is possible like that?

Reviewer: @sfuhrer @Igor-Misic THanks very much. I have fixed it up to say this (I think) in the commit https://github.com/PX4/PX4-user_guide/pull/2226/commits/a3376e34612faff32f1a024c8265de55ebe8108a

There is a bit of a restructure, but the main effect is that it now says this:

> The checks are run when the mission is uploaded and immediately before it is run.
> If any of the checks fail, the user is notified and it is not possible to start the mission.

It also says that if the mode is switched to but not allowed to start due to being infeasible, then this is the same as if there is no mission when you switch to the mode - the vehicle holds/loiters.

Reviewer: Thanks @sfuhrer - I will revert that then. Probably in most cases that will be sufficient - i.e. you tend to boot when you're ready to fly, so that is when making the check makes most sense.

Reviewer: Actually, I see you already reverted.

> The checks involve (selection of most likely ones):
> - First mission item too far away
> - Distance between two subsequent items is too large
> - Any mission item outside of defined geofence
> - More than one land start defined
> - Fixed-wing landing infeasible slope angle
> - Land start item before RTL item
> - Missing takeoff and/or land item (see parameter list below for configuration)

Reviewer: ```suggestion
A subset of the most important checks are listed below:
- First mission item too far away from vehicle
- Distance between two subsequent items is too large
- Any mission item conflicts with a plan or safety geofence
- More than one land start mission item defined ([MAV_CMD_DO_LAND_START](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_LAND_START))
- A fixed-wing landing has an infeasible slope angle
- Land start item (`MAV_CMD_DO_LAND_START`) appears in mission before an RTL item ([MAV_CMD_NAV_RETURN_TO_LAUNCH](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_RETURN_TO_LAUNCH))
- Missing takeoff and/or land item when these are configured as a requirement (see parameter list below for configuration)

```
